### PR TITLE
Add code blocks to the ch16 (get-authenticated-user)

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Resolve `16-get-authenticated-user` in #12

### CODE REVIEW

File: knpu/get-authenticated-user.md [hunk 1]
```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
...
+You can see that "I" is whomever we logged in as. We didn't specify what the username
+should be for that user, but whoever is logged in is who "I" represents.
 
 When we worked with the `ls` scenarios earlier, we needed to share the command output
 string between the steps of a scenario. In this case, we have a similar need: we
 need to share the user object from the step where we log in, with the step where
-"I" author some products. To share data between steps, create a new `private $currentUser;`.
+"I" author some products. To share data between steps, create a new `private $currentUser;`:
+
+[[[ code('100d047bcd') ]]]
```
Comment (bocharsky-bw) [inline knpu/get-authenticated-user.md:77]: @weaverryan Should we add more context here and wrap this property with class in the code block? I show a single line of property declaration here in code block now

File: knpu/get-authenticated-user.md [hunk 2]
```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
...
+
+I already have that relationship setup with in Doctrine. Great!
+
+In `thereAreProducts()`, change the body of this function to `$this->createProducts($count);`:
+
+[[[ code('95e36de658') ]]]
+
+Do the same thing in `iAuthorProducts()` for now:
+
+[[[ code('acd4d6cbd4') ]]]
```
Comment (bocharsky-bw) [inline knpu/get-authenticated-user.md:56]: And the same question, should we wrap this single line with method name like:

``` php
public function iAuthorProducts($count)
{
    $this->createProducts($count);
}
```

Looks like we get it from context above

File: knpu/get-authenticated-user.md [hunk 3]
```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
...
 
-Great - copy the `iAuthorProducts` function code and paste it into our handy `FeatureContext`
-class - near the other product function. These two functions will be similar, so
-we should reuse the logic. Copy the internals of `thereAreProducts`, make a new
-`private function createProducts`. Pass it `$count` as an argument and also an optional
-User object which will be the author for those products. Now, add an if statement
-that says, if `$author` is passed then, `$product->setAuthor()`. I already have that
-relationship setup with in Doctrine. Great!
+```bash
+$ ./vendor/bin/behat features/web/product_admin.feature:13
```
Comment (weaverryan) [inline knpu/get-authenticated-user.md:30]: THANK you for adding commands when we run command - I also try to do this, but I don't catch them all. But (unlike the core Symfony docs) - we don't need to add a `$` - we add a `>` via CSS when it renders :)

File: knpu/get-authenticated-user.md (hunk 2 above)
Comment (weaverryan) [inline knpu/get-authenticated-user.md:56]: Yep, I agree - we should give it more context. Sometimes, we're working in a function, and then we add a new line to it, then another new line later. In these cases, it seems _less_ important to include the "context" on the 2nd and 3rd code-blocks (but still maybe not a bad thing). But when we switch to work on a different function - like we do here - then yea, definitely :). I'm in the code block editor right now, so I'll tweak these.

File: knpu/get-authenticated-user.md (hunk 1 above)
Comment (weaverryan) [inline knpu/get-authenticated-user.md:77]: +1

File: knpu/get-authenticated-user.md [hunk 4]
```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
...
 string between the steps of a scenario. In this case, we have a similar need: we
 need to share the user object from the step where we log in, with the step where
-"I" author some products. To share data between steps, create a new `private $currentUser;`.
+"I" author some products. To share data between steps, create a new `private $currentUser;`:
+
+[[[ code('100d047bcd') ]]]
+
+In `iAmLoggedInAsAnAdmin()`, add `$this->currentUser = $this->thereIsAnAdminUserWithPassword()`:
+
+[[[ code('9d15e806ff') ]]]
```
Comment (weaverryan) [inline knpu/get-authenticated-user.md:81]: I gave this more context too - it was just the single line, I expanded it to show the method name above (and the opening `{` and close `}` after it.

File: knpu/get-authenticated-user.md (hunk 3 above)
Comment (bocharsky-bw) [inline knpu/get-authenticated-user.md:30]: Oh, sorry... I forgot about this CSS feature here.

**Comment by bocharsky-bw** [comment]
@weaverryan Thanks for this review, it was very useful! I got the idea

**Pull request merged**